nemo/collections/asr/models/label_models.py

```diff
@@ -97,29 +108,15 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         self.encoder = EncDecSpeakerLabelModel.from_config_dict(cfg.encoder)
         self.decoder = EncDecSpeakerLabelModel.from_config_dict(cfg.decoder)
 
-        if 'loss' in cfg:
-            if 'angular' in cfg.decoder and cfg.decoder['angular']:
-                logging.info("loss is Angular Softmax")
-                scale = cfg.loss.scale
-                margin = cfg.loss.margin
-                self.loss = AngularSoftmaxLoss(scale=scale, margin=margin)
-            else:
-                logging.info("loss is Softmax-CrossEntropy")
-                if 'weight' in cfg.loss and cfg.loss['weight']:
-                    if cfg.loss.weight == 'auto':
-                        self.cal_labels_occurrence = True
-                        # Goal is to give more weight to the classes with less samples so as to match the ones with the higher frequencies
-                        weight = [
-                            sum(self.labels_occurrence) / (len(self.labels_occurrence) * i)
-                            for i in self.labels_occurrence
-                        ]
-                    else:
-                        weight = cfg.loss.weight
-                    self.loss = CELoss(weight=weight)
-                else:
-                    self.loss = CELoss()
+        if self.labels_occurrence:
+            # Goal is to give more weight to the classes with less samples so as to match the ones with the higher frequencies
+            weight = [sum(self.labels_occurrence) / (len(self.labels_occurrence) * i) for i in self.labels_occurrence]
+
+        if 'loss' not in cfg:
```

Comment:
why is this necessary again? put in else case above

Reply:
it serves to instantiate loss when there is no loss in config file, such as unit test

Reply:
yeah you are right, we can put it as 
```
if self.labels_occurrence:
    weight = [sum(self.labels_occurrence) / (len(self.labels_occurrence) * i) for i in self.labels_occurrence]
else:
    self.loss = CrossEntropyLoss()  # default loss for this class, basically to pass test
```

Reply:
you mean here? 

```
if 'loss' in cfg:
    xxx
else:
    self.loss = CrossEntropyLoss()  # default loss for this class, basically to pass test
    self.eval_loss = CrossEntropyLoss()
```

we cannot assign module before Module.__init__() call